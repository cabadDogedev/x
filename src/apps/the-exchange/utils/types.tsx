import { BridgingProvider, ExchangeOffer } from '@etherspot/prime-sdk/dist/sdk/data';
import { Route } from '@lifi/sdk';

export enum CardPosition {
    SWAP = 'SWAP',
    RECEIVE = 'RECEIVE'
}

export type SwapType = {
    fromAmount: number,
    fromTokenAddress: string,
    fromChainId: number;
    fromTokenDecimals: number;
    toTokenAddress: string,
    toChainId: number,
    toTokenDecimals: number;
    slippage: number;
    fromAccountAddress?: string
    provider?: BridgingProvider;
}

export type SwapOffer = {
    tokenAmountToReceive: number;
    offer: Route | ExchangeOffer;
}

export type AmountType = {
    tokenAmount: number;
    usdAmount: number;
  }
  
export type ChainType = {
    chainId: number;
    chainName: string;
  }
