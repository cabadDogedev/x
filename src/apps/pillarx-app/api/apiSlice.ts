import { createApi, fetchBaseQuery } from '@reduxjs/toolkit/query/react'
import { ApiResponse } from '../../../types/api'

export const pillarXApi = createApi({
    reducerPath: 'pillarXApi',
    baseQuery: fetchBaseQuery({ baseUrl: 'https://feed-nubpgwxpiq-uc.a.run.app/' }),
    endpoints: (builder) => ({
        getTilesInfo: builder.query<ApiResponse, string>({
            query: () => ''
        })
    })
})


export const { useGetTilesInfoQuery } = pillarXApi;
