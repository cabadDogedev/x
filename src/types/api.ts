export type Asset = {
    name: string;
    symbol: string;
    id: number;
    contracts: string[];
    logo: string;
    blockchains: string[];
  }
  
  export type CrossChainBalance = {
    balance: number;
    balanceRaw: string;
    chainId: string;
    address: string;
  }
  
  export type ContractBalance = {
    balance: number;
    balanceRaw: string;
    chainId: string;
    address: string;
    decimals: number;
  }
  
  export type AssetData = {
    asset: Asset;
    realized_pnl: number;
    unrealized_pnl: number;
    allocation: number;
    price: number;
    price_bought: number;
    price_change_24h: number;
    price_change_1h: number;
    total_invested: number;
    min_buy_price: number;
    max_buy_price: number;
    estimated_balance: number;
    token_balance: number;
    cross_chain_balances: {
      [key: string]: CrossChainBalance;
    };
    contracts_balances: ContractBalance[];
  }
  
  export type TotalPnlHistory = {
    realized: number;
    unrealized: number;
  }
  
  export type ApiData = {
    total_wallet_balance: number;
    wallet: string;
    wallets: string[];
    total_realized_pnl: number;
    total_unrealized_pnl: number;
    assets: AssetData[];
    total_pnl_history: {
      [key in '24h' | '7d' | '30d' | '1y']?: TotalPnlHistory;
  };
  
  }
  
  // TO DO - meta type to change when api ready, layout should be enum
  export type Projection = {
    meta: unknown;
    data: ApiData;
    layout: string;
    id: string;
  }
  
  export type ApiResponse = {
    projection: Projection[];
  }
  