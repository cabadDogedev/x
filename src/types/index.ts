export interface AppManifest {
  title: string;
  description: string;
  translations: Record<string, string>;
}
