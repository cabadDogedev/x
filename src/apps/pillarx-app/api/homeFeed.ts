/* eslint-disable no-console */
import { createApi, fetchBaseQuery } from '@reduxjs/toolkit/query/react';
import { addMiddleware } from '../../../store';
import { ApiResponse } from '../../../types/api';

export const homeFeedApi = createApi({
    reducerPath: 'homeFeedApi',
    baseQuery: fetchBaseQuery({ baseUrl:
        process.env.REACT_APP_USE_TESTNETS === 'true' ?
        'https://feed-nubpgwxpiq-uc.a.run.app' :
        'https://feed-7eu4izffpa-uc.a.run.app'
    }),
    endpoints: (builder) => ({
        getTilesInfo: builder.query<ApiResponse, { page: number; address: string }>({
            query: ({ page, address }) => `?page=${page}&address=${address}`,
        }),
    })
})

/**
 * Add this to the store
 */
addMiddleware(homeFeedApi);

export const { useGetTilesInfoQuery } = homeFeedApi;
