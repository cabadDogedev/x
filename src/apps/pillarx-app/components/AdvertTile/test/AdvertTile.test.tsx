import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';

// types
import { Advertisement, ApiLayout, Projection } from '../../../../../types/api';

// components
import AdvertTile from '../AdvertTile';

describe('<AdvertTile />', () => {
    const mockDataAdvert: Projection = {
        meta: {},
        data: {
            slug: 'placement-slug'
        } as Advertisement,
        layout: ApiLayout.AD,
        id: 'id-advert'
    };

    // Unable to test the component further because very dependant on HypeLab SDK

    it('renders correctly and matches snapshot', () => {
        const tree = renderer.create(
            <AdvertTile data={mockDataAdvert} isDataLoading={false} />
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('renders nothing when data is loading', () => {
        render(<AdvertTile data={undefined} isDataLoading={true} />);
        expect(screen.queryByTestId('advert-tile')).toBeNull();
    });
});
