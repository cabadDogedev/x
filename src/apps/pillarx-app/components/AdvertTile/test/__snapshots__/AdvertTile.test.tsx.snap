// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AdvertTile /> renders correctly and matches snapshot 1`] = `null`;
